gtsfm/loader/hilti_loader.py

```diff
@@ -77,29 +88,133 @@ def __init__(
             self._intrinsics[cam_idx] = calibration[0]
             self._cam_T_imu_poses[cam_idx] = calibration[1]
 
+        # Jacobian for transforming covariance
+        self._cam2_J_imu_relative: np.array = self._cam_T_imu_poses[2].AdjointMap()
+
         # Check how many images are on disk.
         self.num_rig_poses: int = self.__get_num_rig_poses()
         if self._max_length is not None:
             self.num_rig_poses = min(self.num_rig_poses, self._max_length)
 
+        # Read GT poses from odom file:
+        self._fastlio_poses = self.__read_fastlio_poses()
+
         # Read the constraints from the lidar/constraints file
-        self.constraints = self.__load_constraints()
-        logger.info("Number of constraints: %d", len(self.constraints))
+        all_constraints = self._load_constraints()
+        filtered_constraints = list(self._covariance_filtered_constraints(all_constraints))
+        # TODO(Frank): below does *not* give same result if iterable above not converted to list!
```

Comment:
what was going on here? is it easy to fix? @dellaert ?

Reply:
Casted to list in the filter function